compiler/rustc_const_eval/src/interpret/operand.rs

```diff
@@ -730,14 +730,43 @@ impl<'mir, 'tcx: 'mir, M: Machine<'mir, 'tcx>> InterpCx<'mir, 'tcx, M> {
                 // Return the cast value, and the index.
                 (discr_val, index.0)
             }
-            TagEncoding::Niche { untagged_variant, ref niche_variants, niche_start } => {
+            TagEncoding::Niche { untagged_variant, ref niche_variants, niche_start, ref flag } => {
+                let is_magic_val = if let Some(flag) = flag {
+                    let flag_val = self.read_immediate(&self.operand_field(op, flag.field)?)?;
+                    let flag_val = flag_val.to_scalar();
+                    match flag_val.try_to_int() {
+                        Err(dbg_val) => {
+                            // So this is a pointer then, and casting to an int
+                            // failed. Can only happen during CTFE. If the magic
+                            // value is 0 and the scalar is not null, we know
+                            // this is OK. Anything else we conservatively
+                            // reject.
+                            let ptr_valid =
```

Comment:
```suggestion
                            // the pointer cannot be the magic value. Anything else
                            // we conservatively reject.
                            let ptr_definitely_not_magic_val =
```
plus renaming the variable in its two uses below

(`ptr_valid` is a bad name also in the later code for niche handling, but it's much more confusing here IMO. if you want to rename it to `ptr_definitely_not_in_niche_variants` that would be great)